hooks/gen_docs/gen_docs_env_vars.py

```diff
@@ -0,0 +1,385 @@
+"""Generates the documentation on KPOps environment variables."""
+
+import csv
+import shutil
+from collections.abc import Callable
+from dataclasses import dataclass
+from pathlib import Path
+from textwrap import fill
+from typing import Any
+
+from pytablewriter import MarkdownTableWriter
+from typer.models import ArgumentInfo, OptionInfo
+from typing_extensions import Self
```

Comment:
technically the preferred way would be to try importing from std lib and have typing_extensions as a fallback

```python
try:
    from typing import Self
except ImportError:
    from typing_extensions import Self
```

but it's fine either way in this case since it's a dev-only module